src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCauseBuildAction.java

```diff
@@ -342,7 +343,17 @@ private static void adjustProjectDisplayName(
     private static Set<Run<?, ?>> getDownstreamBuilds(
             final Run build) {
 
-        Set<Run<?, ?>> foundDbf = new TreeSet<Run<?, ?>>();
+        Set<Run<?, ?>> foundDbf = new TreeSet<Run<?, ?>>(new Comparator<Run<?, ?>>() {
+            @Override
+            public int compare(Run<?, ?> o1, Run<?, ?> o2) {
+                final int res = o1.getParent().getFullName().compareTo(o2.getParent().getFullName());
```

Comment:
Well, perhaps there is a need for having it in correct order. What matters to me is making sure there are no risk for `NullPointerException`s. You for example don't even test if `o1` or `o2` are `null`.

And then there is `getParent()` where the risk is very low that it would return `null` and `getFullName()` which is also low risk. But if any of them are it will be very difficult to fix from a stack trace since you can't tell which one is null of all those three calls on the same line.